### PR TITLE
refactor: Migrate CallingCell to React

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: src/script/components/list/ConversationListCallingCell.tsx [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+  multitasking,
+  maximizedTileVideoParticipant,
+  videoSpeakersActiveTab,
+  videoGrid,
+  hasAccessToCamera,
+  isSelfVerified,
+  teamRepository,
+  callingRepository,
+}) => {
+  const callState = container.resolve(CallState);
```
Comment (thisisamir98) [inline src/script/components/list/ConversationListCallingCell.tsx]: can be a prop so for testing later it can be mocked maybe?

**Review by thisisamir98 — APPROVED** [review]

File: src/script/components/list/ConversationListCallingCell.test.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
+/*
+ * Wire
+ * Copyright (C) 2019 Wire Swiss GmbH
```
Comment (bennycode) [inline src/script/components/list/ConversationListCallingCell.test.ts:3]: 2021

Comment (bennycode) [inline src/script/components/list/ConversationListCallingCell.test.ts]:
```suggestion
 * Copyright (C) 2021 Wire Swiss GmbH
```

File: src/script/components/list/ConversationListCallingCell.test.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    expect(callingCellPage.getOutgoingLabel().exists()).toBe(true);
+  });
+  it('displays a call that is connecting', async () => {
+    const props = await createProps();
+    props.call.state(CALL_STATE.ANSWERED);
+    const callingCellPage = new ConversationListCallingCellPage(props);
+
+    expect(callingCellPage.getConnectingLabel().exists()).toBe(true);
+  });
+  it('displays the running time of an ongoing call', async () => {
```
Comment (bennycode) [inline src/script/components/list/ConversationListCallingCell.test.ts:112]: please add a line break after `it` blocks

**Review by bennycode — CHANGES_REQUESTED** [review]

File: src/script/components/list/ConversationListCallingCell.test.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    currentAvailableDeviceId: {
+      audioOutput: ko.pureComputed(() => 'test'),
+    },
+  } as MediaDevicesHandler);
+  call.state(state);
+  return call;
+};
+
+const createProps = async () => {
+  const testFactory = new TestFactory();
```
Comment (Yserz) [inline src/script/components/list/ConversationListCallingCell.test.ts]: Is it possible to do this without the TestFactory?

File: src/script/components/list/ConversationListCallingCell.test.ts (hunk 3 above)
Comment (ffflorian) [reply src/script/components/list/ConversationListCallingCell.test.ts:112]: ~~I'm against adding an extra line break after `it` blocks. We never do that.~~
I misunderstood. All good.

File: src/script/components/list/ConversationListCallingCell.tsx [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+                    <div
+                      className="call-ui__button call-ui__button--red call-ui__button--large"
+                      onClick={() => (isIncoming ? callActions.reject(call) : callActions.leave(call))}
+                      title={t('videoCallOverlayHangUp')}
+                      data-uie-name="do-call-controls-call-decline"
+                    >
+                      <Icon.Hangup className="small-icon" style={{maxWidth: 17}} />
+                    </div>
+                  )}
+                  {isIncoming && !isDeclined && (
```
Comment (ffflorian) [inline src/script/components/list/ConversationListCallingCell.tsx]: We already checked for `!isDeclined`

```suggestion
                  {isIncoming && (
```

**Review by ffflorian — APPROVED** [review]

File: src/script/components/list/ConversationListCallingCell.tsx [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+                    .map(participant => (
+                      <ParticipantItem
+                        key={participant.clientId}
+                        participant={participant.user}
+                        hideInfo
+                        noUnderline
+                        callParticipant={participant}
+                        selfInTeam={selfUser?.inTeam()}
+                        isSelfVerified={isSelfVerified}
+                        external={teamRepository.isExternal(participant.user.id)}
```
Comment (Yserz) [inline src/script/components/list/ConversationListCallingCell.tsx]: I think `isExternal` could be extracted to the `TeamState`. Thus we could use the `TeamState` and prevent using the whole `TeamRepository`.

**Review by ffflorian — APPROVED** [review]

**Review by Yserz — APPROVED** [review]

**Pull request merged**